lua/weapons/weapon_laserpointer.lua

```diff
@@ -0,0 +1,599 @@
+end
+
+function SWEP:Deploy()
+	if(CLIENT)then self.HornBG = nil end
+	if(CLIENT and IsValid(self:GetOwner()))then self.ViewModelFOV = self:GetOwner():GetFOV() end
+	if(SERVER and self:GetOwner().laserpointerlastcolor)then self:SetColorMode(self:GetOwner().laserpointerlastcolor) end
+	return true
+end
+
+function SWEP:Holster()
+	return true
+end 
+
+
+function SWEP:Reload()
+	
+	return true
+end
+
+hook.Add( "KeyPress", "LaserColorPicker", function( ply, key )
+if(key == IN_RELOAD and IsValid(ply:GetActiveWeapon()) and ply:GetActiveWeapon():GetClass() == "weapon_laserpointer")then
+ply:GetActiveWeapon():ChangeColor()
+ply:GetActiveWeapon():ButtonSound(true)
+end
+end )
+
+hook.Add( "KeyRelease", "LaserColorPicker", function( ply, key )
+if((key == IN_RELOAD or key == IN_ATTACK2) and IsValid(ply:GetActiveWeapon()) and ply:GetActiveWeapon():GetClass() == "weapon_laserpointer")then
+ply:GetActiveWeapon():ButtonSound(false)
+end
+end )
+
+
+function SWEP:ChangeColor()
+
+local nextmode = self:GetColorMode() + 1
+if(self.ColorFunctions[nextmode] == nil)then nextmode = 0 end
+self:SetColorMode(nextmode)
+self:GetOwner().laserpointerlastcolor = self:GetColorMode()
+end
+
+
+function SWEP:PrimaryAttack( )
+	if(self:Clip1() <= 0)then return true end
+		if(self:GetOnState() != true)then
+			local can = hook.Call("PlayerCanHaveLaserBeams",self:GetOwner(),self)
+			if(can == false)then self:SetBeamMode(false) end
+			self:SetOnState(true)
+			self:ButtonSound(true)
+		end
+		local ply = self:GetOwner()
+		local trace = ply:GetEyeTrace()
+		local take = 1 
+		if(self:GetBeamMode())then take = 4 end
+		self:SetClip1(math.max(self:Clip1() - take,0))
+		if(self:GetBeamMode())then LaserPointer_SVBeam(ply,self,ply:EyePos(),ply:GetAimVector()) end
+			timer.Create( self:EntIndex().."LaserPointerOff", 0.3+FrameTime(), 1, function()
+				if(IsValid(self))then
+					self:SetOnState(false)
+					self:ButtonSound(false)
+				end
+			end)
+	self:SetNextPrimaryFire(CurTime()+0.2)
+end
+
+hook.Add( "KeyRelease", "LaserPointerReleaseCheck", function( ply, key )
+    if ( key == IN_ATTACK ) then
+		local pointer = ply:GetActiveWeapon()
+		if(IsValid(pointer) and pointer:GetClass() == "weapon_laserpointer" and pointer:GetOnState() == true)then
+			pointer:SetOnState(false)
+			pointer:SetNextPrimaryFire(CurTime()+0.2)
+			pointer:ButtonSound(false)
+			timer.Destroy( pointer:EntIndex().."LaserPointerOff")
+		end
+    end
+end )
+
+hook.Add("HUDPaintBackground","laserpointer_blinding",function()
+
+end)
+
+function SWEP:OnDrop()
+
+end
+
+local pony_head_hitbones = {}
+pony_head_hitbones["LrigScull"] = true
+pony_head_hitbones["Mane01"] = true
+pony_head_hitbones["Mane02"] = true
+pony_head_hitbones["Mane03"] = true
+pony_head_hitbones["Mane04"] = true
+pony_head_hitbones["Mane05"] = true
+
+local laser_material = Material( "sprites/physg_glow1" )
+local beam_material = Material( "sprites/light_glow02_add" )
+laser_material:SetVector("$color2",Vector(4,4,4))
```

Comment:
The problem with this is it changes the material parameters anywhere else "physg_glow1" or "light_glow02_add" is used. Maybe CreateMaterial() ?